File: data/gpt-4o/readable_format_HPEC/fortran/file028.f90
program DRB079_taskdep3_orig_no
    use omp_lib
    implicit none

    integer :: i, j, k
    i = 0

    !$omp parallel
        !$omp single
            !$omp task depend (out:i)
                call sleep(3)
                i = 1
            !$omp end task
            !$omp task depend (in:i)
                j = i
            !$omp end task
            !$omp task depend (in:i)
                k = i
            !$omp end task
        !$omp end single
    !$omp end parallel

    print 100, j, k
    100 format ('j =',i3,2x,'k =',i3)

    if (j /= 1 .and. k /= 1) then
        print*, 'Race Condition'
    end if

end program

File: data/gpt-4o/readable_format_HPEC/cpp/file028.cpp
#include <stdio.h>
#include <unistd.h>
#include <omp.h>

int main()
{
    int i = 0, j, k;
#pragma omp parallel
    {
#pragma omp single
        {
#pragma omp task depend(out : i)
            {
                sleep(3);
                i = 1;
            }
#pragma omp task depend(in : i)
            {
                j = i;
            }
#pragma omp task depend(in : i)
            {
                k = i;
            }
        }
    }
    printf("j = %d  k = %d\n", j, k);
    if (j != 1 || k != 1)
    {
        printf("Race Condition\n");
    }
    return 0;
}
